pandas/io/sql.py

```diff
@@ -964,7 +964,13 @@ def insert_data(self) -> tuple[list[str], list[np.ndarray]]:
 
         for i, (_, ser) in enumerate(temp.items()):
             if ser.dtype.kind == "M":
-                d = ser.dt.to_pydatetime()
+                if isinstance(ser._values, ArrowExtensionArray):
+                    with warnings.catch_warnings():
+                        warnings.filterwarnings("ignore")
+                        # GH#52459 to_pydatetime will return Index[object]
+                        d = ser.dt.to_pydatetime()
```

Comment:
Probably good to call `np.asarray` on d ahead of the deprecation enforcement since `data_list` is typed `list[np.ndarray]`